### ISSUE TITLE
[Bug]: Dropdown doesn't add accessible name to listbox

### ISSUE THREAD

**Issue by RamyRafik** [microsoft/fluentui#29384]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (20) x64 Intel(R) Xeon(R) W-2255 CPU @ 3.70GHz
    Memory: 32.54 GB / 63.69 GB
  Browsers:
    Edge: Spartan (44.22621.2283.0), Chromium (117.0.2045.35)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codesandbox.io/s/ngllrm?file=/example.tsx

## Actual Behavior
Although Dropdown has an `aria-labeledby` prop, it's only applied to the Button w/ `role="combobox"`, but not the Div w/ `role=listbox`

![image](https://github.com/microsoft/fluentui/assets/20221435/290eb7ca-58cf-4b2c-8964-67f390cbc050)

## Expected Behavior
An Element w/ `role=listbox` should have an Accessible Name.
https://accessibilityinsights.io/info-examples/web/aria-input-field-name/

### Requested priority

High

### Products/sites affected

Microsoft Dynamics 365

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
